utils/hct/hctdb.py

```diff
@@ -5508,9 +5511,37 @@ def UFI(name, **mappings):
         )
         next_op_idx += 1
 
+        # RayQuery
+        self.add_dxil_op(
+            "AllocateRayQuery2",
+            next_op_idx,
+            "AllocateRayQuery2",
+            "allocates space for RayQuery and return handle",
+            "v",
+            "",
+            [
+                db_dxil_param(0, "i32", "", "handle to RayQuery state"),
+                db_dxil_param(
+                    2,
+                    "u32",
+                    "constRayFlags",
+                    "Valid combination of RAY_FLAGS",
+                    is_const=True,
+                ),
+                db_dxil_param(
+                    3,
+                    "u32",
+                    "constRayQueryFlags",
+                    "Valid combination of RAYQUERY_FLAGS",
+                    is_const=True,
+                ),
+            ],
+        )
+        next_op_idx += 1
+
         # End of DXIL 1.8 opcodes.
         self.set_op_count_for_version(1, 8, next_op_idx)
-        assert next_op_idx == 258, (
+        assert next_op_idx == 259, (
             "258 is expected next operation index but encountered %d and thus opcodes are broken"
```

Comment:
Maybe?

```suggestion
            "259 is expected next operation index but encountered %d and thus opcodes are broken"
```

Reply:
258 has been reserved by Simon for AllocateRayQuery2, so that opcode should be all set, no change needed.

Reply:
I don't understand why the `258` in the message doesn't also need to be updated to `259`, like in the assertion expression.

Reply:
Ah I get it, you're right.